odoo/addons/base/models/res_bank.py

```diff
@@ -100,10 +100,7 @@ def _get_supported_account_types(self):
     country_code = fields.Char(related='partner_id.country_code', string="Country Code")
     note = fields.Text('Notes')
 
-    _unique_number = models.Constraint(
-        'unique(sanitized_acc_number, partner_id)',
-        "The combination Account Number/Partner must be unique.",
-    )
+    _unique_number = models.UniqueIndex("(sanitized_acc_number, partner_id) WHERE active IS TRUE")
```

Comment:
We should keep the nice message when the constraint fails

Also need to double check if this requires an upgrade script